xarray/core/dataset.py

```diff
@@ -5914,5 +5921,169 @@ def pad(
 
         return self._replace_vars_and_dims(variables)
 
+    def idxmin(
+        self,
+        dim: Hashable = None,
+        axis: int = None,
```

Comment:
Generally I don't think we offer `axis` as a arg except to maintain compatibility, but no harm in including

Reply:
I was trying to keep the arguments similar to`min`, `max`, `argmin`, and `argmax`

Reply:
`axis` is really confusing and doesn't really fit in with xarray. Let's not add it?

Reply:
@dcherian: it is already present in `min`, `max`, `argmin`, and `argmax`.  Since these all serve similar purposes, shouldn't we keep them as consistent as possible?

Reply:
I think it's in those functions for compatibility with numpy, but I don't think it's in there intentionally. I'm not adamant either way, probably -0.2

Reply:
Okay, I removed it.